src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -108,7 +108,7 @@ public function panelAction(Request $request, $token)
         }
 
         if (!$profile = $this->profiler->loadProfile($token)) {
-            return new Response($this->twig->render('@WebProfiler/Profiler/info.html.twig', array('about' => 'no_token', 'token' => $token, 'request' => $request)), 200, array('Content-Type' => 'text/html'));
+            throw new NotFoundHttpException(sprintf('Token "%s" was not found in the database.', $token));
```

Comment:
This changes the behavior a little bit, as you break out of the profiler layout for an unknown token (which seems consistent with an unknown ?panel= param btw.)

I think it's negligible :)